src/couch_replicator_connection.erl

```diff
@@ -171,6 +175,33 @@ delete_worker(Worker) ->
     ok.
 
 
+-spec incr_monitor(pid(), #state{}) -> #state{}.
+incr_monitor(Owner, State) ->
+    NewMonitors = case dict:find(Owner, State#state.monitors) of
+        error ->
+            MRef = monitor(process, Owner),
+            dict:store(Owner, {1, MRef}, State#state.monitors);
```

Comment:
Monitors should stack. So when it relinquishes A it would demonitor RefA. Then still be monitored for B, C by RefB and RefC. The nice thing would be that monitor refs match 1-to-1 with workers and perhaps we won't have to keep another data structure to map 1 owner to workers by Pid.

I tried it as a quick example:
```erlang
> P1=spawn(fun() -> receive ok -> ok end end).
> R1=erlang:monitor(process, P1).
> R2=erlang:monitor(process, P1).
> erlang:demonitor(R1).
> exit(P1, kill).
> flush().
Shell got {'DOWN',#Ref<0.0.3.69>,process,<0.48.0>,killed}
ok

> R2.
#Ref<0.0.3.69>
```
So we monitored and demonitored P1 using R1 monitor (demonitoring there is like relinquishing), but R2 was still left and would correctly detect death of P1.